File: src/main/c/inline_method.cpp
#include <cassert>
#include <vector>
#include <iostream>

using namespace std;

// Before
class Calculator
{
private:
    vector<int> array;
public:
    Calculator(int num, ...);
    bool isOverThousand(int val);
    int compute();
};

Calculator::Calculator(int num, ...)
{
    va_list args;
    va_start(args, num);

    for (int i = 0; i < num; i++) {
        int val = va_arg(args, int);
        array.push_back(val);
    }

    va_end(args);
}

bool Calculator::isOverThousand(int val)
{
    return val > 1000;
}

int Calculator::compute()
{
    int total = 0;
    for (int i = 0; i < array.size(); i++) {
        total += array[i];
    }

    if (isOverThousand(total)) {
        total = 1000;
    }

    return total;
}

// After
class Calculator2
{
private:
    vector<int> array;
public:
    Calculator2(int num, ...);
    bool isOverThousand(int val);
    int compute();
};

Calculator2::Calculator2(int num, ...)
{
    va_list args;
    va_start(args, num);

    for (int i = 0; i < num; i++) {
        int val = va_arg(args, int);
        array.push_back(val);
    }

    va_end(args);
}

int Calculator2::compute()
{
    int total = 0;
    for (int i = 0; i < array.size(); i++) {
        total += array[i];
    }

    if (total > 1000) {
        total = 1000;
    }

    return total;
}


int
main(void)
{
    Calculator c(1, 2, 31);
    Calculator c2(1, 2, 31);

    assert(c.compute() == c2.compute());
}

File: src/main/python/inline_method.py
## Before
class Calculator(object):
    __slots__ = ['__values']

    def __init__(self, *args):
        self.__values = args

    def compute(self):
        total = sum(self.__values)
        if self.is_over_thousand(total):
            total = 1000
        return total

    def is_over_thousand(self, value):
        return value >= 1000

## After
class Calculator2(object):
    """ Method is_over_thousand move in inline code"""
    __slots__ = ['__values']

    def __init__(self, *args):
        self.__values = args

    def compute(self):
        total = sum(self.__values)
        if total >= 1000:
            total = 1000
        return total


if __name__ == '__main__':
    calculator = Calculator(1, 5, -10, 2, 100)
    calculator2 = Calculator2(1, 5, -10, 2, 100)

    assert calculator.compute() == calculator2.compute()
